languages.toml

```diff
@@ -368,7 +388,7 @@ source = { git = "https://github.com/tree-sitter/tree-sitter-c", rev = "7175a6dd
 name = "cpp"
 scope = "source.cpp"
 injection-regex = "cpp"
-file-types = ["cc", "hh", "c++", "cpp", "hpp", "h", "ipp", "tpp", "cxx", "hxx", "ixx", "txx", "ino", "C", "H", "cu", "cuh", "cppm"]
+file-types = ["cc", "hh", "c++", "cpp", "hpp", "h", "ipp", "tpp", "cxx", "hxx", "ixx", "txx", "ino", "C", "H", "cu", "cuh", "cppm", "h++", "ii", "inl", "hpp.in", "h.in"]
```

Comment:
I believe we don't currently support the `hpp.in`/`h.in` kind of extension. Rust's [`Path::extension`](https://doc.rust-lang.org/std/path/struct.Path.html#method.extension) splits the final extension (basically rsplit_once rather than split_once). It might be nice to support these though and it would be useful for Erlang `.app.src` extensions too, so we could consider changing this in syntax.rs in the future

Reply:
I think that could just be a suffix or be covered by #8006

Reply:
>I think that could just be a suffix
Is this something that's supported today or should I remove them for this PR?

Reply:
cc @the-mikedavis

Reply:
Yeah, for now let's switch them to `{ suffix = ".hpp.in" }, { suffix = ".h.in" }` etc

Reply:
@the-mikedavis done

Reply:
<img width="659" alt="Screenshot 2023-10-02 at 00 24 55" src="https://github.com/helix-editor/helix/assets/14347895/3434b1bf-e4a5-4764-8150-f46b3a65963d">
@the-mikedavis this seems to trigger an error

Reply:
That looks like an error message from the language server, I doubt it's related to the changes here